### ISSUE TITLE
Bog Witch update breaks games graphics settings

### ISSUE THREAD

**Issue by Graphic-Shark** [SpikeHimself/XPortal#105]
**Describe the bug**
When starting up valheim, the graphic settings are reverted to a previous state and any attempt to modify the graphics tab will cause the game to crash. Even going to the graphics tab and hitting "OK" will break the game. All the text on the menus are also broken. 

**To Reproduce**
before going into your world and selecting an avatar, go to settings > Graphics and try to change anything. then just hit "OK" at the bottom. The screen will go white and the game window will crash showing only the console window for mods with an error read for unity.

**Expected behaviour**
Change graphic settings like resolution without it crashing. 

**LogOutput file(s)**
Please have `Debug` output enabled. Attach the file called `LogOutput.log` which can be found in the directory `Valheim\BepInEx\`.
If you play on a server, include the server's `LogOutput.log` as well.

**Version**
[LogOutput.log](https://github.com/user-attachments/files/17560692/LogOutput.log)
1.2.21

**Environment**

* Singleplayer

**Other mods**
Only happens with XPortal

**Priority**

* Medium (XPortal kind of still works, but it's not pleasant to use)

**Additional information**
This is what shows in the console:

![Screenshot (1)](https://github.com/user-attachments/assets/ba3be873-21a9-4841-97f0-32227547f541)

[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
Localization.GetBoundKeyString (System.String bindingName, System.Boolean emptyStringOnMissing) (at <18ff35ff344d40f994ffbe8686d655cf>:0)
Localization.Translate (System.String word) (at <18ff35ff344d40f994ffbe8686d655cf>:0)
Localization.Localize (System.String text) (at <18ff35ff344d40f994ffbe8686d655cf>:0)
Localization.Localize (UnityEngine.Transform root) (at <18ff35ff344d40f994ffbe8686d655cf>:0)
Localize.Start () (at <18ff35ff344d40f994ffbe8686d655cf>:0)
[LogOutput.log](https://github.com/user-attachments/files/17560758/LogOutput.log)

**Comment by SpikeHimself** [comment]
Hi @Graphic-Shark, thank you for reporting this issue!

While I acknowledge and accept that installing XPortal ultimately triggers this situation, the actual root cause is in Jotunn. XPortal uses a functionality in Jotunn which is currently breaking the game.

I cannot fix this until Jotunn releases a new version with a fix.

Relevant part from your logs:

```
MissingMethodException: Method not found: ZInput/ButtonDef .ZInput.AddButton(string,GamepadInput,single,single,bool,bool)
Stack trace:
Jotunn.Managers.InputManager+Patches.RegisterCustomInputs (ZInput __instance) (at D:/Projects/Jotunn/JotunnLib/Managers/InputManager.cs:179)
```

**Comment by Graphic-Shark** [comment]
Thanks for the quick reply! Looks like Jotunn will update in a few days after the Bog Witch goes live. Hopefully, that gets everything stitched back up.

**Comment by SpikeHimself** [comment]
Fixed in Jotunn v2.21.2

https://github.com/Valheim-Modding/Jotunn/releases/tag/v2.21.2
